boa/src/builtins/array/array_iterator.rs

```diff
@@ -0,0 +1,149 @@
+    ///  - [ECMA reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-createarrayiterator
+    pub(crate) fn create_array_iterator(
+        ctx: &Context,
+        array: Value,
+        kind: ArrayIterationKind,
+    ) -> Result<Value> {
+        let array_iterator = Value::new_object(Some(
+            &ctx.realm()
+                .environment
+                .get_global_object()
+                .expect("Could not get global object"),
+        ));
+        array_iterator.set_data(ObjectData::ArrayIterator(Self::new(array, kind)));
+        array_iterator
+            .as_object_mut()
+            .expect("array iterator object")
+            .set_prototype_instance(ctx.iterator_prototypes().array_iterator());
+        Ok(array_iterator)
+    }
+
+    /// %ArrayIteratorPrototype%.next( )
+    ///
+    /// Gets the next result in the array.
+    ///
+    /// More information:
+    ///  - [ECMA reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-%arrayiteratorprototype%.next
+    pub(crate) fn next(this: &Value, _args: &[Value], ctx: &mut Context) -> Result<Value> {
+        if let Value::Object(ref object) = this {
+            let mut object = object.borrow_mut();
+            if let Some(array_iterator) = object.as_array_iterator_mut() {
+                let index = array_iterator.next_index;
+                if array_iterator.array.is_undefined() {
+                    return Ok(create_iter_result_object(ctx, Value::undefined(), true));
+                }
+                let len = array_iterator
+                    .array
+                    .get_field("length")
+                    .as_number()
+                    .ok_or_else(|| ctx.construct_type_error("Not an array"))?
+                    as i32;
+                if array_iterator.next_index >= len {
+                    array_iterator.array = Value::undefined();
+                    return Ok(create_iter_result_object(ctx, Value::undefined(), true));
+                }
+                array_iterator.next_index = index + 1;
+                match array_iterator.kind {
+                    ArrayIterationKind::Key => {
+                        Ok(create_iter_result_object(ctx, Value::integer(index), false))
+                    }
+                    ArrayIterationKind::Value => {
+                        let element_value = array_iterator.array.get_field(index);
+                        Ok(create_iter_result_object(ctx, element_value, false))
+                    }
+                    ArrayIterationKind::KeyAndValue => {
+                        let element_value = array_iterator.array.get_field(index);
+                        let result = Array::make_array(
+                            &Value::new_object(Some(
+                                &ctx.realm()
+                                    .environment
+                                    .get_global_object()
+                                    .expect("Could not get global object"),
```

Comment:
```suggestion
                                ctx.global_object()
```
we can call the `Context::global_object()` to get the global object instead of searching through the environment